nodejs/eks/cluster.ts

```diff
@@ -223,14 +225,38 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
         throw new Error("instanceRole and instanceRoles are mutually exclusive, and cannot both be set.");
     }
 
-    // If no VPC was specified, use the default VPC.
+    if (args.subnetIds && (args.publicSubnetIds || args.privateSubnetIds)) {
+        throw new Error("subnetIds, and the use of publicSubnetIds and/or privateSubnetIds are mutually exclusive. Choose a single approach.");
+    }
+
+    // Configure default networking architecture.
     let vpcId: pulumi.Input<string> = args.vpcId!;
     let subnetIds: pulumi.Input<pulumi.Input<string>[]> = args.subnetIds!;
-    if (args.vpcId === undefined) {
+    const publicSubnetIds: pulumi.Input<pulumi.Input<string>[]> = args.publicSubnetIds!;
+    const privateSubnetIds: pulumi.Input<pulumi.Input<string>[]> = args.privateSubnetIds!;
+    let clusterSubnetIds: pulumi.Input<pulumi.Input<string>[]> = [];
+
+    // If no VPC is set, use the default VPC's subnets.
+    if (!args.vpcId) {
         const invokeOpts = { parent: parent };
         const vpc = aws.ec2.getVpc({ default: true }, invokeOpts);
         vpcId = vpc.then(v => v.id);
         subnetIds = vpc.then(v => aws.ec2.getSubnetIds({ vpcId: v.id }, invokeOpts)).then(subnets => subnets.ids);
+        clusterSubnetIds = subnetIds;
+    }
+
+    // Form the subnetIds to use on the cluster from either:
+    //  - subnetIds
+    //  - A combination of privateSubnetIds and/or publicSubnetIds.
+    if (args.subnetIds) {
+        clusterSubnetIds = subnetIds;
```

Comment:
It looks wrong that this uses `args.subnetIds` for the check but `subnetIds` in the assignment.

I honestly can't quite follow how this logic intersects with the population of `subnetIds` in the default VPC case above.

Reply:
> It looks wrong that this uses args.subnetIds for the check but subnetIds in the assignment.

I've updated the checks, PTAL.

> I honestly can't quite follow how this logic intersects with the population of subnetIds in the default VPC case above.

See comment [above](https://github.com/pulumi/pulumi-eks/pull/239#discussion_r320446888).